console/console-init/ui/src/modules/device/components/AddCredential/AddCredential.tsx

```diff
@@ -0,0 +1,215 @@
+import { PageSection } from "@patternfly/react-core";
+import {
+  CredentialList,
+  ICredential,
+  IExtension,
+  ISecret
+} from "modules/device/components";
+import { uniqueId, findIndexByProperty } from "utils";
+
+export const AddCredential: React.FC<{}> = () => {
+  const getExtensionsFieldsInitialState = () => {
+    const initialState: IExtension = {
+      id: uniqueId(),
+      parameter: "",
+      type: "",
+      value: ""
+    };
+    return initialState;
+  };
+
+  const getCredentialsFieldsInitialState = () => {
+    const initialState: ICredential = {
+      id: uniqueId(),
+      "auth-id": "",
+      type: "",
+      secrets: [{}],
+      extensions: [getExtensionsFieldsInitialState()],
+      enabled: true,
+      isExpandedAdvancedSetting: false
+    };
+    return initialState;
+  };
+
+  const [credentials, setCredentials] = useState([
+    getCredentialsFieldsInitialState()
+  ]);
+  const [type, setType] = useState();
+  const [activeCredentialFormId, setActiveCredentialFormId] = useState();
+
+  const getSecretsFieldsInitialState = () => {
+    let initialState: ISecret = {};
+    switch (type && type.toLowerCase()) {
+      case "hashed_password":
+        initialState = {
+          "pwd-hash": "",
+          "pwd-function": "",
+          salt: "",
+          "not-before": "",
+          "not-after": "",
+          comment: ""
+        };
+        break;
+      case "x509":
+        initialState = { "not-before": "", "not-after": "", comment: "" };
+        break;
+      case "psk":
+        initialState = {
+          key: "",
+          "not-before": "",
+          "not-after": "",
+          comment: ""
+        };
+        break;
+      default:
+        break;
+    }
+    return initialState;
+  };
+
+  const getFormInitialStateByProperty = (property: string) => {
+    let initialState = {};
+    if (property === "secrets") {
+      initialState = getSecretsFieldsInitialState();
+    } else if (property === "extensions") {
+      initialState = getExtensionsFieldsInitialState();
+    } else if (property === "credentials") {
+      initialState = getCredentialsFieldsInitialState();
+    }
+    return initialState;
+  };
+
+  const setSecretsInitialFormState = () => {
+    const newCredentials = [...credentials];
+    const index = findIndexByProperty(
+      credentials,
+      "id",
+      activeCredentialFormId
+    );
+    if (index >= 0) {
+      const initialState = getFormInitialStateByProperty("secrets");
+      newCredentials[index]["secrets"] = [{ id: uniqueId(), ...initialState }];
+      setCredentials(newCredentials);
+    }
+  };
+
+  useEffect(() => {
+    setSecretsInitialFormState();
+  }, [type]);
+
+  const onSelectType = (id: string, event: any, value: string) => {
+    setType(value);
+    setActiveCredentialFormId(id);
+    handleInputChange(id, event, value);
+  };
+
+  const addMoreItem = (id?: string, property?: string) => {
+    let newCredentials: any[];
```

Comment:
Can this be a type of ICredential[]?

Reply:
I tried to use ICredential[] but some index issue/error was coming that's why i used any[]. I'll take a look when I will work on container part.